sdl_android/src/main/java/com/smartdevicelink/streaming/RTPH264Packetizer.java

```diff
@@ -0,0 +1,488 @@
+ *
+ * $ gst-launch-1.0 souphttpsrc location=http://127.0.0.1:5050 ! "application/x-rtp-stream" ! rtpstreamdepay ! "application/x-rtp,media=(string)video,clock-rate=90000,encoding-name=(string)H264" ! rtph264depay ! "video/x-h264, stream-format=(string)avc, alignment=(string)au" ! avdec_h264 ! autovideosink sync=false
+ */
+
+/**
+ * This class receives H.264 byte stream (in Annex-B format), parses it, construct RTP packets
+ * from it based on RFC 6184, then frame the packets based on RFC 4571.
+ * The primary purpose of using RTP is to carry timestamp information along with the data.
+ *
+ * @author Sho Amano
+ */
+public class RTPH264Packetizer extends AbstractPacketizer implements IEncoderListener, Runnable {
+
+	// Approximate size of data that mOutputQueue can hold in bytes.
+	// By adding a buffer, we accept underlying transport being stuck for a short time. By setting
+	// a limit of the buffer size, we avoid buffer overflows when underlying transport is too slow.
+	private static final int MAX_QUEUE_SIZE = 256 * 1024;
+
+	private static final int FRAME_LENGTH_LEN = 2;
+	private static final int MAX_RTP_PACKET_SIZE = 65535;  // because length field is two bytes (RFC 4571)
+	private static final int RTP_HEADER_LEN = 12;
+	private static final byte DEFAULT_RTP_PAYLOAD_TYPE = 96;
+	private static final int FU_INDICATOR_LEN = 1;
+	private static final int FU_HEADER_LEN = 1;
+	private static final byte TYPE_FU_A = 28;
+
+	// To align with StreamPacketizer class
+	private final static int TLS_MAX_RECORD_SIZE = 16384;
+	private final static int TLS_RECORD_HEADER_SIZE = 5;
+	private final static int TLS_RECORD_MES_AUTH_CDE_SIZE = 32;
+	private final static int TLS_MAX_RECORD_PADDING_SIZE = 256;
+
+	private final static int MAX_DATA_SIZE_FOR_ENCRYPTED_SERVICE =
+			TLS_MAX_RECORD_SIZE - TLS_RECORD_HEADER_SIZE - TLS_RECORD_MES_AUTH_CDE_SIZE- TLS_MAX_RECORD_PADDING_SIZE;
+
+	private boolean mServiceProtected;
+	private Thread mThread;
+	private BlockingQueue<ByteBuffer> mOutputQueue;
+	private volatile boolean mPaused;
+	private boolean mWaitForIDR;
+	private NALUnitReader mNALUnitReader;
+	private byte mPayloadType = 0;
+	private int mSSRC = 0;
+	private char mSequenceNum = 0;
+	private int mInitialPTS = 0;
+
+	/**
+	 * Constructor
+	 *
+	 * @param streamListener The listener which this packetizer outputs SDL frames to
+	 * @param serviceType The value of "Service Type" field in SDL frames
+	 * @param sessionID The value of "Session ID" field in SDL frames
+	 * @param session The SdlSession instance that this packetizer belongs to
+	 */
+	public RTPH264Packetizer(IStreamListener streamListener,
+	                         SessionType serviceType, byte sessionID, SdlSession session) throws IOException {
+
+		super(streamListener, null, serviceType, sessionID, session);
+
+		mServiceProtected = session.isServiceProtected(_serviceType);
+
+		if (bufferSize == 0) {
+			// fail safe
+			bufferSize = MAX_DATA_SIZE_FOR_ENCRYPTED_SERVICE;
```

Comment:
This should be the agreed upon MTU for the video streaming service

Reply:
Thanks, implemented with https://github.com/smartdevicelink/sdl_android/pull/592/commits/135a684668340311697bf49ab0b8b61c09c5e1e1.